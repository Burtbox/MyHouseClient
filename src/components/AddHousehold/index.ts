import AddHousehold, { AddHousehold as DisconnectedAddHousehold } from './AddHousehold';

export { DisconnectedAddHousehold };
export default AddHousehold;
