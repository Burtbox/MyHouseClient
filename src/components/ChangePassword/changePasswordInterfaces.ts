import { IComponentProps } from '../../interfaces/componentInterfaces';
import { IUser } from '../Users/usersInterfaces';

export interface IChangePasswordProps extends IComponentProps {
    loggedInUser: IUser;
}

export interface IChangePasswordState {
    passwordUpdate: {
        userId: string,
        CURRENTPASSWORD: string,
        NEWPASSWORD: string,
        NEWPASSWORDCONFIRM: string, // ED! need to sort this out as part of the user interfaces tidy up
    };
    passwordUpdating: boolean;
    passwordUpdated: boolean;
}
