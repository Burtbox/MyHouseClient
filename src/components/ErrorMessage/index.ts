import ErrorMessage, { ErrorMessage as DisconnectedErrorMessage } from './ErrorMessage';

export default ErrorMessage;

export { DisconnectedErrorMessage };
