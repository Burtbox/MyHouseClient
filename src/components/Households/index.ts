import Households, { Households as DisconnectedHouseholds } from './Households';

export default Households;

export { DisconnectedHouseholds };
