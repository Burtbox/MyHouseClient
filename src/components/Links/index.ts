import Links, { Links as DisconnectedLinks } from './Links';

export default Links;
export { DisconnectedLinks };
