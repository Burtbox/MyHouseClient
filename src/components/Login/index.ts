import Login, { Login as DisconnectedLogin } from './Login';

export default Login;

export { DisconnectedLogin };
