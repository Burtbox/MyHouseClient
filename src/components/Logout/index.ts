import Logout, { Logout as DisconnectedLogout } from './Logout';

export default Logout;

export { DisconnectedLogout };
