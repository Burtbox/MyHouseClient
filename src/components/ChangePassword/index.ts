import ChangePassword, { ChangePassword as DisconnectedChangePassword } from './ChangePassword';

export default ChangePassword;

export { DisconnectedChangePassword };
