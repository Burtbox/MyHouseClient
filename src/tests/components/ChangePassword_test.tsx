import * as React from 'react';
import { shallow } from 'enzyme';
import { DisconnectedChangePassword } from '../../components/ChangePassword';
import { IChangePasswordProps } from '../../components/ChangePassword/changePasswordInterfaces';
import loggedInUser from '../shared/loggedInUser';

describe('Change Password test suite', () => {
    test('Render Change Password component', () => {
        const props: IChangePasswordProps = {
            loggedInUser,
            dispatch: null,
            history: null,
        };
        const ChangePasswordScreen = shallow(
            <DisconnectedChangePassword {...props} />,
        );
        expect(ChangePasswordScreen).toMatchSnapshot();
    });
});
