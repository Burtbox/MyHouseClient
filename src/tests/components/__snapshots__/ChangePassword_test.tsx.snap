// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Change Password test suite Render Change Password component 1`] = `
<form
  onSubmit={[Function]}
  style={
    Object {
      "marginTop": "20px",
      "textAlign": "center",
    }
  }
>
  <h2>
    Change Password
  </h2>
  <div>
    <TextField
      disabled={false}
      floatingLabelFixed={false}
      floatingLabelText="Current Password"
      fullWidth={false}
      hintText="************"
      maxlength="50"
      multiLine={false}
      name="CURRENTPASSWORD"
      onChange={[Function]}
      required={true}
      rows={1}
      type="password"
      underlineShow={true}
      value=""
    />
  </div>
  <div>
    <TextField
      disabled={false}
      floatingLabelFixed={false}
      floatingLabelText="New Password"
      fullWidth={false}
      hintText="************"
      maxlength="50"
      multiLine={false}
      name="NEWPASSWORD"
      onChange={[Function]}
      required={true}
      rows={1}
      type="password"
      underlineShow={true}
      value=""
    />
  </div>
  <div>
    <TextField
      disabled={false}
      floatingLabelFixed={false}
      floatingLabelText="Confirm Password"
      fullWidth={false}
      hintText="************"
      maxlength="50"
      multiLine={false}
      name="NEWPASSWORDCONFIRM"
      onChange={[Function]}
      required={true}
      rows={1}
      type="password"
      underlineShow={true}
      value=""
    />
  </div>
  <FlatButton
    disabled={false}
    fullWidth={false}
    label="Update"
    labelPosition="after"
    labelStyle={Object {}}
    onKeyboardFocus={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onTouchStart={[Function]}
    primary={false}
    secondary={false}
    type="submit"
  />
  <Snackbar
    autoHideDuration={4000}
    message="Password updated"
    onRequestClose={[Function]}
    open={false}
  />
</form>
`;
